### ISSUE TITLE
Mapping turn lanes for navigation in 30 cities in US

### ISSUE THREAD

**Issue by abhisheksaikia** [mapbox/mapping#180]
**Objective:** Map missing [turn lanes](http://wiki.openstreetmap.org/wiki/Key:turn:lanes) from secondary and up in 30 US cities.

![turnlanes](https://cloud.githubusercontent.com/assets/8401827/15009897/ca49c6c0-1206-11e6-99ad-a458e6d7539c.jpeg)

**Team:** @mapbox/team-data

**Features to map**: Turn lanes (motorway, trunk, primary and secondary) from satellite imagery

**Focus Area**: 30 top cities in US as per population
### Available data sources

| Source | Description | Quality | Coverage | License |
| --- | --- | --- | --- | --- |
| Mapbox Satellite | Imagery | Intermediate | Good | :white_check_mark: |
| Bing Imagery | Imagery | High | Good | :white_check_mark: |
| USGS large scale imagery | Imagery | High | limited | :white_check_mark: |
### Tasks

| Location | Start | End(ETA) | Validation |
| --- | --- | --- | --- |
| [San Francisco](http://tasks.openstreetmap.us/project/35) | 05/5/2016 | 05/11/2016 | Done |
| [Portland](http://tasks.openstreetmap.us/project/36) | 05/16/2016 | 05/17/2016 | Done |
| [Dallas & Fort Worth](http://tasks.openstreetmap.us/project/37) | 05/18/2016 | 5/23/2016 | Done |
| [New York](http://tasks.openstreetmap.us/project/39) | 05/23/2016 | 05/30/2016 | Done |
| [Detroit](http://tasks.openstreetmap.us/project/40) | 05/30/2016 | 06/02/2016 | Done |
| [Houston](http://tasks.openstreetmap.us/project/42) | 06/03/2016 | 06/09/2016 | Done |
| [Philadelphia](http://tasks.openstreetmap.us/project/44) | 06/08/2016 | 06/15/2016 | Done |
| [Baltimore](http://tasks.openstreetmap.us/project/45) | 06/15/2016 | 06/16/2016 | Done |
| [Oklahoma](http://tasks.openstreetmap.us/project/46) | 06/22/2016 | 06/22/2016 | Done |
| [Boston](http://tasks.openstreetmap.us/project/47) | 06/22/2016 | 06/27/2016 | Done |
| [Denver](http://tasks.openstreetmap.us/project/48) | 06/28/2016 | 06/29/2016 | Done |
| [Seattle](http://tasks.openstreetmap.us/project/49) | 06/29/2016 | 07/01/2016 | Done |
| [Chicago](http://tasks.openstreetmap.us/project/50) | 07/01/2016 | 08/05/2016 | Done |
| [San Diego](http://tasks.openstreetmap.us/project/51) | 08/08/2016 | 08/10/2016 | Done |
| [Austin](http://tasks.openstreetmap.us/project/55) | 08/10/2016 | 08/11/2016 | Done |
| [Phoenix](http://tasks.openstreetmap.us/project/56) | 08/11/2016 | 08/18/2016 | Done |
| [DC](http://tasks.openstreetmap.us/project/74) | 23/01/2017 |30/01/2017|Done|
| [LA_exterior](http://tasks.openstreetmap.us/project/81) | 13/02/2017 | | |
#### Setup
- Activate the following in from JOSM preferences > Map Settings
  - **Map Paint Styles**: Lanes and road attributes
  - **Tagging Presets**: lane attributes
- Filter out non road features with `boundary: | leisure: | landuse: | waterway: | amenity: | natural: | building:`
- Add inverse filter: `highway=motorway|trunk|primary|secondary|*link|motorway_link or untagged`
- Read http://wiki.openstreetmap.org/wiki/Key:turn:lanes
- Read https://github.com/mapbox/mapping/wiki/Mapping-guide-for-turn-lanes-from-imagery
- Install the 🔪  tool in JOSM to easily split the roads: https://wiki.openstreetmap.org/wiki/JOSM/Plugins/Knife-tool

#### Mapping
- Check the roads for turn lanes at exits and entry points
  ![](https://cloud.githubusercontent.com/assets/126868/11710074/b58b52e4-9f42-11e5-971a-000a699a0b6d.png)
- Split the road into segments based on the total number of lanes for the segment
- For each road segment tag the turn lanes based on the road markings in the direction of the way. Eg. `turn:lanes=left|through|right`. See [OSM turn:lanes](http://wiki.openstreetmap.org/wiki/Key:turn:lanes)
- For bi-directional segments, use `turn:lanes:backward=` and `turn:lanes:forward=` to specify turn lanes in each direction.
- Verify correct tagging using the `Lane attributes` style
  ![](https://cloud.githubusercontent.com/assets/126868/11713668/d3f6d498-9f5a-11e5-91ea-5d9ab5714b39.png)
#### Issues to keep in mind:
- For QA and edge cases in turn lanes mapping and how to handle them**(parking lanes, splitting roads for start and end of turn lanes, handling of dual carriageways, etc)** look into the [QA for turn lane data](https://github.com/mapbox/mapping/wiki/QA-for-turn-lane-data) document.

**Changeset Comment**: `Add missing turn lanes from satellite imagery https://github.com/mapbox/mapping/issues/180`

**Source**: `Bing/Mapbox/USGS Large Scale Imagery`

**Turn lane mapping tickets**: #144, #153

**Comment by abhisheksaikia** [comment]
Capturing the initial estimates for turn lanes mapping sprint in top 30 US cities:

According to the statistics that we generated during the [LA turn lanes mapping sprint](https://github.com/mapbox/mapping/issues/153#issuecomment-204369279), this is the estimated timeline for completing the turn lanes mapping sprint in the full 30 cities:

**Total area to be covered in 30 US cities**: 37,545 square kilometers
**Estimated features to map**: *~375,450 turn lanes in 30 US cities(~10 turn lanes per square kilometer)*
**Turn lanes per person per hour**:*~25*
**Total number of team members working full time(6 hours per day)**: *15* 
**Total estimated working days**: *~100 working days(5 months)*

#### Optimizing the workflow and initial time estimates:

* These statistics are heavily dependent upon our [LA turn lanes mapping sprint](https://github.com/mapbox/mapping/issues/153). LA had a very high density of turn lanes in even primary, secondary and tertiary roads which is generally not the case in other cities. This will immediately lead to a reduction in our mapping effort.

* Optimizing the JOSM toolset to help map turn lanes better: a **knife tool** for quick splitting of road segments in turn lanes mapping, a JOSM plugin that automatically sets the first lane as `left` and also automatically predicts the turn lane scheme for the next junction, once a junction has been tagged.

* This time estimate also does not include the amount of time taken for validation. As 30 cities will be a huge mapping project, we need to find faster and better ways of validating turn lanes than the traditional method of validating tasks in the tasking manager.

#### Next actions:

* [ ] @planemad to work with @samely and @Rub21 and create the turn lanes mapping tools to optimize our mapping effort.
* [ ] @jothirnadh and me to capture and update daily statistics for turn lanes mapping in each city and consequently review our timeline based on the mapping speed by our team for the first 2 cities (Portland and San Francisco).

cc @mapbox/team-data

**Comment by abhisheksaikia** [comment]
The statistics for total number of `turn:lanes` added  each day by the @mapbox/team-data during this sprint will be updated in this [spreadsheet](https://docs.google.com/spreadsheets/d/1mszzwfAHSIzAol5X9sQ-Dko8FYYIhVwOcZ2UUhA1yRA/edit#gid=1249134110) for each of the 30 cities. Please update the `start` and `end` dates once a city is completed.

**Comment by abhisheksaikia** [comment]
@mapbox/team-data have completed adding [turn lanes in SF Bay area](http://tasks.openstreetmap.us/project/35).

#### [Final Statistics](https://docs.google.com/spreadsheets/d/1mszzwfAHSIzAol5X9sQ-Dko8FYYIhVwOcZ2UUhA1yRA/edit#gid=1249134110) in SF bay area turn lanes mapping sprint(excluding validation):

* Total number of working days: 4
* Average number of mappers working per day: 12 (average 6 hours per day)
* Total number of personal hours by the team: 288
* Total number of turn lanes mapped: 10,977
* Turn lanes/mapper/hour: ~ 38 turn lanes 
* Total existing turn lanes in SF bay in OSM: 15386
* Total area covered in SF bay: 1925.66 square kilometers
* Turn lanes density in San Francisco Bay area: ~8 turn lanes/kilometer

#### Observations:

* The mapping speed for @mapbox/team-data has increased from an [estimated **25 turn lanes** per mapper per hour](https://github.com/mapbox/mapping/issues/180#issuecomment-217369515) to **38 turn lanes** per mapper per hour.

* The Turn lanes density in SF is ~8 turn lanes/sq. km, down from our [initial estimate of 10 turn lanes per square kilometer](https://github.com/mapbox/mapping/issues/180#issuecomment-217369515).

* This will immediately lead to a significant reduction in our initial mapping estimate of 100 working days for 30 cities. The estimate can be reviewed based on the average of density of turn lanes we get from the first three cities we map.

**Comment by abhisheksaikia** [comment]
As part of our discussion with @planemad @maning @jothirnadh on optimization of workflow for adding turn lanes, the following changes are made to the workflow which we will follow while mapping [turn lanes in Portland](http://tasks.openstreetmap.us/project/36) and all further tasks in the turn lanes mapping sprint.

* We are going to map all turn lanes from `secondary and up` instead of `tertiary and up`. 
* Use the [knife tool](https://github.com/Rub21/auto-tools) for quick splitting of ways: [Steps for activating the Knife tool](https://gist.github.com/abhisheksaikia/e4bd79661e1d39b617319c2f6d206789)

cc @mapbox/team-data

**Comment by jothirnadh** [comment]
> Portland City coverage: missing highways, turn lanes coverage, satellite imagery coverage.

I did an overview of the roads mapped in Portland City. The city seems to be well mapped and the quality of the satellite imagery (Bing) is good for mapping turn lanes. 

CC:/ @abhisheksaikia @maning

**Comment by jothirnadh** [comment]
Updating the [statistics in SF bay](https://github.com/mapbox/mapping/issues/180#issuecomment-218694863) (Stats excluding validation) from above, and providing a [team wise breakup](https://docs.google.com/spreadsheets/d/1mszzwfAHSIzAol5X9sQ-Dko8FYYIhVwOcZ2UUhA1yRA/edit#gid=0) of the mapping stats between India and Peru teams:

Team |  Average hours per mapper/day | Average number of mappers per day | Turn lanes added by  team | Turn lane per mapper/hr
--- | --- | --- | --- | --- | ---
Bangalore | 5 | 6.25  | 2071 | ~17
Peru | 8 | 5.75 | 8252 | ~45 

The statistics of turn lanes added during the SF bay sprint: 

* Turn lanes added by the Peru and India teams combined during the sprint: 10,323
* **Turn lanes added by the community during the sprint: 674**

cc:/ @mapbox/team-data

**Comment by abhisheksaikia** [comment]
Updating the filter to include all roads until secondary (including all road links), according to our [revised workflow](https://github.com/mapbox/mapping/issues/180#issuecomment-218717074): `highway=motorway|trunk|primary|secondary|*link|motorway_link`

cc @mapbox/team-data

**Comment by oini** [comment]
We're skipping adding turn lanes to situations where the lanes are marked by yellow lines on both sides. These are probably turn-lanes-bothways.

![screen shot 2016-05-16 at 4 12 49 pm](https://cloud.githubusercontent.com/assets/1933377/15287587/8dd24396-1b81-11e6-8527-b84814804175.png)

![screen shot 2016-05-16 at 4 13 00 pm](https://cloud.githubusercontent.com/assets/1933377/15287591/9165c50a-1b81-11e6-9521-93cea4b7be16.png)

**Comment by oini** [comment]
Another Observation: Portland has pretty unclear satellite imagery (both Bing and USGS Large Scale Imagery).

![screen shot 2016-05-16 at 4 24 56 pm](https://cloud.githubusercontent.com/assets/1933377/15287799/0370cb76-1b83-11e6-9d61-dde4a75e1905.png)

**Comment by jothirnadh** [comment]
The 🔪  tool is updated with the auto select way option as per this [issue](https://github.com/Rub21/auto-tools/issues/3). Please do update your plugin with the following [jar file](https://github.com/Rub21/auto-tools/raw/split/dist/auto_tools.jar) and restart your JOSM to see the changes.

cc:/ @mapbox/team-data

**Comment by jothirnadh** [comment]
## Portland Task Completion Summary

We wrapped up [mapping turn lanes in the Portland area](http://tasks.openstreetmap.us/project/36#) and added a total of 2,823 turn lanes. The approach to mapping in Portland was [different from the work we did in San Francisco](https://github.com/mapbox/mapping/issues/180#issuecomment-218694863), and we only covered roads that were classfied as secondary and above. Here is a full summary of our approach, and results. We welcome feedback on our approach, and suggestions for improving this workflow.

#### Pre-mapping Checklist

* We looked at **satellite imagery coverage** and assessed for clarity as this significantly impacts one's ability to clearly identify turnlanes :100:
* We assessed the existing **road network coverage** to determine the number of roads under mapped :100:
* We researched the presence of **unmapped dual-carriageways** to determine the extent to which roads need to be splitted so as to avoid creating conflicts when mapping turn-lanes 👍 

#### Mapping roads classified as secondary and above

We decided to focus on adding turn lanes on roads classified as secondary or above after our research showed that a majority of turn lanes are present on such roads, and tertiary roads and residential streets have minimal to no turn lanes to map. This also helped us keep the scope of the mapping effort more focused. 

#### [Final Statistics](https://docs.google.com/spreadsheets/d/1mszzwfAHSIzAol5X9sQ-Dko8FYYIhVwOcZ2UUhA1yRA/edit#gid=0) in Portland turn lanes mapping sprint (excluding validation):

* Total number of working days: **1**
* Number of mappers worked: **16 (average 6.5 hours per day)**
* Total number of people hours by the team: **104**
* Existing turn lanes before sprint: **193**
* Total number of turn lanes mapped during sprint: **2,823**
* Turn lanes/mapper/hour: **~24 turn lanes** 
* Total area covered: **500.38 square kilometers**
* Turn lanes density: **~6 turn lanes/kilometer**

#### Observations:

The overall mapping speed for adding turn lanes in Portland decreased from our experience [mapping turn lanes in Los Angeles city](https://github.com/mapbox/mapping/issues/153#issuecomment-204369279). This decrease can be attributed to a higher density of turn lanes present in LA which made it faster to identify turn-lanes overall.

### Going Forward

* we're kicking off [adding turn lanes in the cities of Dallas and Forth Worth, Texas](http://tasks.openstreetmap.us/project/37)
* we're kicking off researching the coverage of turn lanes, satellite imagery quality, and road network in New York city

We'll be posting details here, and welcome whomever is interested to join us in mapping turn lanes :)

**Comment by dobratzp** [comment]
Especially when doing large-scale editing of roads, be careful when splitting and combining Way objects.  These Way objects may be members of road route relation and/or public transport route relations.  The order of the Way objects within the Relation is significant.

**Comment by planemad** [comment]
@dobratzp this is a really serious issue since we do get conflicts on route relation members when working across cities. Sadly the JOSM relation conflict mechanism does not make it very easy to properly resolve them and I suspect relations worldwide are breaking everyday since we do not yet have a usable conflict management and integrity checking interface for all relations on the map.

From our end, the team tries to be really careful when working on roads with relations, but we're lacking a good QA workflow to ensure nothing is broken once the task is complete. For starters, we will ticket out suggestions to improve the intelligence of conflict management in JOSM so that it can suggest a resolution. Would be great to hear your thoughts as well on how this could be done.

cc @mapbox/team-data

**Comment by dobratzp** [comment]
I agree that it seems like JOSM doesn't always do the right thing when you are splitting Ways as far as bus routes and turn restrictions go.  If all I had to do was fix a few bus routes by making them contiguous again, it wouldn't be so bad.

I started the process of fixing Portland area bus routes, but then I realized just how bad [Changeset 39375345](https://www.openstreetmap.org/changeset/39375345) is really is.  There are few duplicate Way objects, which are fairly easy to remedy.  However, there are a number of relations which are completely broken.

I'm not sure how you even managed to do this.  Somehow, you've replaced the members of all of the following relations with the same members from [Bus Route 52: Portland Community College Rock Creek => Beaverton Transit Center](https://www.openstreetmap.org/relation/4132040)

- https://www.openstreetmap.org/relation/1654457
- https://www.openstreetmap.org/relation/166138
- https://www.openstreetmap.org/relation/1829927
- https://www.openstreetmap.org/relation/1830136
- https://www.openstreetmap.org/relation/2026211
- https://www.openstreetmap.org/relation/2903404
- https://www.openstreetmap.org/relation/2903405
- https://www.openstreetmap.org/relation/4033810
- https://www.openstreetmap.org/relation/4040351
- https://www.openstreetmap.org/relation/4071514
- https://www.openstreetmap.org/relation/4071570
- https://www.openstreetmap.org/relation/4100472
- https://www.openstreetmap.org/relation/4100473
- https://www.openstreetmap.org/relation/4132040
- https://www.openstreetmap.org/relation/4241959
- https://www.openstreetmap.org/relation/4241962
- https://www.openstreetmap.org/relation/4244274
- https://www.openstreetmap.org/relation/4250113
- https://www.openstreetmap.org/relation/5553728

Some of these are bus route relations and some of them are turn restriction relations.

Also, all of the members of [Bus 52: Beaverton Transit Center => Portland Community College Rock Creek](https://www.openstreetmap.org/relation/4132039) have been deleted.

Unfortunately, I can't cleanly revert [Changeset 39375345](https://www.openstreetmap.org/changeset/39375345), so I may have to ask for help with the Data Working Group.

**Comment by dobratzp** [comment]
Before I got to the mess of [Changeset 39375345](https://www.openstreetmap.org/changeset/39375345), I did want to mention a few JOSM warning that you should definitely be paying attention to as they relate specifically to turn lanes:

### Number of lanes greater than *:lanes
For example, if you have lanes=2, then you should have 2 items in your various turn:lanes tags

### oneway=yes together with turn:lanes:forward
Use turns:lanes and not turns:lanes:forward for oneway streets

**Comment by dobratzp** [comment]
I'm wondering if end up having multiple people editing different portions of the same route relation at the same time and you are getting conflicts.  I would suggest making changeset that are **much smaller** than what you are currently doing:

1. Download a short segment of road
1. Edit the turn lanes
1. Upload
1. Restart JOSM

**Comment by planemad** [comment]
@dobratzp Looking into https://www.openstreetmap.org/changeset/39375345 its really strange, would be useful to know how it might have happened, @poornibadrinath any ideas? 

**Validation**
- Number of lanes greater than *:lanes. We leave the lanes tag untouched and only focus on turn:lanes. The reasoning is that for a routing system the `turn:lane` information would override the lane information when present since they may not always be consistent as in this case. Its also an open question if turn lanes should be considered as a [full width traffic lane as per the OSM wiki](http://wiki.openstreetmap.org/wiki/Key:lanes). This is also easily fixable via an automated edit.
- oneway=yes together with turn:lanes:forward. This is taken care of in our validation workflow. I just fixed 24 such cases using [this overpass query](http://overpass-turbo.eu/s/gkY).

**Avoiding relation conflicts**
Making changesets will not completely eliminate the issue since two members working on some part of a long route relation is still quite high. The only way to guarantee no conflicts is for no one else to edit in the same city simultaneously. This is currently a weakness in OSM and affects a lot of HOT projects as well. The long term solution is smarter conflict resolution on the software side since it is extremely easy for a human to break relations with the current tools, even for advanced mappers.

**Comment by dobratzp** [comment]
It seems like there is some inconsistency in how you are tagging turn lanes.  Sometimes you use "none" and sometimes not.  For example:
`||left` vs. `none|none|left`
Are these values equivalent for turn:lanes?  If so, please standardize on one for consistency.

**Comment by osMatt** [comment]
While I was looking for some recently added left turns at intersections, I noticed a wrong turn lane tagging. It should be possible to check for these systematically with an overpass query. http://www.openstreetmap.org/way/419223674 is tagged with `||left|`. Pattern to check: the turn lane string does not start with left, but there is some left turn after a pipe symbol. It is very unlikely to have a left turn lane that is to the right of a lane that is not tagged at a left turn lane.

**Comment by abhisheksaikia** [comment]
@dobratzp I have run a overpass query to determine the number of turn lanes that have `none` in the tag and it comes out to 278 in Portland.The query is: http://overpass-turbo.eu/s/gnF. Both `left||` and `left|none|none` are used widely as seen here: http://taginfo.openstreetmap.org/keys/?key=turn%3Alanes&filter=ways#values. Also the `Lane and road attributes` style appears to take both these styles of tagging as equivalent. But thanks for bringing this up. From now on we will standardise this for consistency and stick to the style of `left||` instead of `left|none|none`.

cc @mapbox/team-data

**Comment by abhisheksaikia** [comment]
@osMatt I have run a [overpass query](http://overpass-turbo.eu/s/gnP) in Dallas for all the turn lanes containing `left` but not starting with `left`. At present it shows all the turn lanes that does not start with `l`. It can be modified to include only those turn lanes which start with `|`. For now, it's showing 196 suspicious ways, which will be eyeballed and validated during our validation work. Thanks for pointing this out. This and other common errors like `oneway-yes` with `turn:lanes:forward` and `turn:lanes:`backward` is definitely something we need to validate and correct en masse once each city is completed. Have opened a ticket documenting these issues here: https://github.com/osmlab/osmlint/issues/109

**Comment by abhisheksaikia** [comment]
## Dallas Task Completion Summary

We wrapped up [mapping turn lanes in Dallas and Fort Worth](http://tasks.openstreetmap.us/project/37#) and added and validated a total of **6,267** turn lanes in Dallas. The total turn lanes count in Dallas after the sprint is 6622 at present. Similar to our approach in [Portland](https://github.com/mapbox/mapping/issues/180#issuecomment-220346553) we only focused on adding turn lanes to roads classified secondary and above. 

#### Improved Tooling

We regularly used the [knife tool](https://github.com/Rub21/auto-tools/raw/split/dist/auto_tools.jar) for quick splitting of ways while adding turn lanes. [Give it a try](https://gist.github.com/abhisheksaikia/0c9e47592b6a1f5fdebabbba4d561a4b) and share any feedback you have for improving it.

#### Community participation

We got a lot of help from OpenStreetMap contributors for this sprint, especially when it involved validating the data we added. Here are a few examples of this: 

* There was a significant issue of [breaking of turn restrictions and bus routes relations in Portland](https://github.com/mapbox/mapping/issues/180#issuecomment-220530881) raised by @dobratzp. A [separate ticket](https://github.com/mapbox/mapping/issues/185) was created for this cleanup and @dobratzp reverted the broken relations to their previous versions.

We have tightened up our validation work based on feedback received to clean up two common errors en masse: 
  - [`Oneway=yes` mixed with `turn:lanes:forward` and `turn:lanes:backward`](https://github.com/mapbox/mapping/issues/180#issuecomment-220531645) by using this [overpass query](http://overpass-turbo.eu/s/gkY) for cleanup.
  - [Wrong `left` turn lane tagging](https://github.com/mapbox/mapping/issues/180#issuecomment-220844337): The suspicious `turn:lanes` which have a `left` after a `|, through or right` can be cleaned up by using this [overpass query](http://overpass-turbo.eu/s/gnP)

#### Observations

* The satellite imagery quality in Dallas and Fort Worth was not of the highest quality for adding turn lanes. For this purpose, imagery had to be alternated frequently between Bing, Mapbox and USGS satellite imagery for verification and mapping missing turn lanes.
* The density of turn lanes was the lowest among all the three cities mapped till now: ~2.5 turn lanes per square kilometer but this advantage was offset by the relatively poor image quality and the need to eyeball a huge area (~2,653 square kilometers)

#### Going forward
The team will continue adding [turn lanes in New York](http://tasks.openstreetmap.us/project/39#). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by jothirnadh** [comment]
### New York task completion summary:
We wrapped up mapping turn lanes in [New York](http://tasks.openstreetmap.us/project/39#) and added a total of 7,509 turn lanes. The total turn lanes count in New York after the sprint is 7780 at present. Similar to our approach in [Dallas and Fort Worth](http://tasks.openstreetmap.us/project/37#) we only focused on adding turn lanes to roads classified as secondary and above.

#### Community participation

As per the community suggestions in our [Portland task](https://github.com/mapbox/mapping/issues/185) we tried to be more concrete wihile mapping turn lanes on the roads with pre existing relations.

#### Final Statistics in New York turn lanes mapping sprint (including validation):

* Total number of working days: **6**
* Existing turn lanes before sprint: **49**
* Total number of turn lanes mapped during sprint: **7,731**
* Total area covered: **1827 square kilometers**
* Turn lanes density: **~4.5 turn lanes/ square kilometer**

#### Observations
* The Satellite imagery coverage in New York is good enough to map turn lanes. We stuck to Bing imagery for the entire task.
* There are on an average of ~4.5 turn lanes per square kilometer in New York which is higher than the turn lanes in [Dallas](https://github.com/mapbox/mapping/issues/180#issuecomment-221273357).
* We encountered many missing highway links during the process of mapping turn lanes which can be covered in further mapping projects.

#### Going forward
The team will continue adding [turn lanes in Detroit](http://tasks.openstreetmap.us/project/40#). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by jothirnadh** [comment]
### Detroit task completion summary:
We wrapped up mapping turn lanes in [Detroit](http://tasks.openstreetmap.us/project/40#) and added a total of 2,409 turn lanes. The total turn lanes count in Detroit after the sprint is 2,634 at present. Similar to our approach in [New York](http://tasks.openstreetmap.us/project/39#) we only focused on adding turn lanes to roads classified as secondary and above.

#### Final Statistics in Detroit turn lanes mapping sprint (including validation):

* Total number of working days: **4**
* Existing turn lanes before sprint: **225**
* Total number of turn lanes mapped during sprint: **2,409**
* Total area covered: **1125.22 square kilometers**
* Turn lanes density: **~2 turn lanes/ square kilometer**

#### Observations
* The Satellite imagery coverage in Detroit is good enough to map turn lanes. We stuck to Bing imagery for the entire task.
* There are on an average of ~2 turn lanes per square kilometer in Detroit which is lower than the turn lanes in other cities that we mapped so far.

#### Going forward
The team will continue adding [turn lanes in Houston](http://tasks.openstreetmap.us/project/42#). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by jothirnadh** [comment]
### Houston task completion summary:
We wrapped up mapping turn lanes in [Houston](http://tasks.openstreetmap.us/project/42#) and added a total of 9,133 turn lanes. The total turn lanes count in Houston after the sprint is 9,248 at present. Similar to our approach in [Detroit](http://tasks.openstreetmap.us/project/40#) we only focused on adding turn lanes to roads classified as secondary and above.

#### Final Statistics in Detroit turn lanes mapping sprint (including validation):

* Total number of working days: **5**
* Existing turn lanes before sprint: **115**
* Total number of turn lanes mapped during sprint: **9,133**
* Total area covered: **3503.25 square kilometers**
* Turn lanes density: **~3 turn lanes/ square kilometer**

#### Observations
* The Satellite imagery coverage in Houston is not good enough to map turn lanes. We switched between Bing and USGS large scale imagery to map turn lanes for better guidance. This led us to invest a bit more time than usual to add a proper turn lane.
* There are on an average of ~3 turn lanes per square kilometer in Houston which is lower than the average turn lanes in other cities that we mapped so far.

#### Going forward
The team will continue adding [turn lanes in Philadelphia](http://tasks.openstreetmap.us/project/44#). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by abhisheksaikia** [comment]
@rickmastfan67 Thank you for pointing the issues out. We are constantly looking for feedback from the community on our mapping ticket. It was an oversight on our part to not capture the validation workflow with to-fix on this ticket. We are presently making sure that the [edge cases documented by you are incorporated into the to fix task](https://github.com/osmlab/osmlint/issues/109#issuecomment-225330660) and will also make sure that all future validation work through to-fix will be properly documented and ticketed in public in the mapping repo.

cc @mapbox/team-data

**Comment by Andygol** [comment]
@rickmastfan67

I highly appreciate your hard work on improving map data in your area. 
But you allow yourself to judge tags values on your own. All values (`none`, `through`, `left`, `right` and so on) of `turn:lanes` are valid but not all combination of them.

I've just re-read again description of  [`turn:lanes`](http://wiki.openstreetmap.org/wiki/Key:turn). Value `none` is used only in this cases

>- There are **no turn indications** on this lane.
>- Used especially for **better readability** in case of many lanes

Based on this ☝️,  combination of `none;slight_right` **isn't** valid.

For purposes that you described have to be used tag [`transit:lanes` ](http://wiki.openstreetmap.org/wiki/Key:transit), see example on Wiki-OSM - http://wiki.openstreetmap.org/wiki/Key:turn#Motorway_with_links_and_destinations 

👇 

<img width="688" alt="key turn openstreetmap wiki 2016-06-13 15-39-12" src="https://cloud.githubusercontent.com/assets/369696/16007762/05211240-317d-11e6-9450-9218627a76d6.png">

I am so sorry that you have spent so much time to indicate places where roads are branching using a false picture of the application values of tag `turn:lanes`. 

It would be great if you consider the usage of another tag such as `transit:lanes` for this purpose!

/cc @planemad @Rub21 @jothirnadh @abhisheksaikia

**Comment by planemad** [comment]
@rickmastfan67 do you think you can create an overpass query for the lanes you might have tagged like 👆 We could try migrating them to `transit:lanes`.

**Comment by jothirnadh** [comment]
### Philadelphia and Baltimore task completion summary:
We wrapped up mapping turn lanes in [Philadephia](http://tasks.openstreetmap.us/project/44#) & [Baltimore](http://tasks.openstreetmap.us/project/45#) and added a total of 2,823 turn lanes in Philadephia and 1,438 turn lanes in Baltimore. The total turn lanes count in Philadelphia after the sprint is 3,024 and 1,918 in Baltimore. Similar to our previous mapping approach we only focused on adding turn lanes to roads classified as secondary and above.

#### Final Statistics in Philadelphia turn lanes mapping sprint (including validation):

* Total number of working days: **5**
* Existing turn lanes before sprint: **178**
* Total number of turn lanes mapped during sprint: **2,823**
* Total area covered: **893.09 square kilometers**
* Turn lanes density: **~3 turn lanes/ square kilometer**

#### Final Statistics in Baltimore turn lanes mapping sprint (including validation):

* Total number of working days: **2**
* Existing turn lanes before sprint: **480**
* Total number of turn lanes mapped during sprint: **1,438**
* Total area covered: **671.21 square kilometers**
* Turn lanes density: **~3 turn lanes/ square kilometer**

#### Observations
* The satellite imagery(Bing) in both the cities is good enough to map turn lanes.
* There are on an average of ~3 turn lanes/sq km in both the cities. 

#### Going forward
The team will continue adding [turn lanes in Oklahoma](http://tasks.openstreetmap.us/project/46#). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by abhisheksaikia** [comment]
## Oklahoma and Boston Task Completion Summary

We wrapped up mapping turn lanes in [Oklahoma](http://tasks.openstreetmap.us/project/46#) and [Boston](http://tasks.openstreetmap.us/project/47#). We have added and validated a total of **1613** turn lanes in Oklahoma and a total of **872** turn lanes in Boston. The total turn lanes count in Oklahoma and Boston after the sprint is **3226** and **911** at present.  

We have also introduced an plugin called [TurnLanes-tagging plugin](https://www.mapbox.com/blog/turnlanes-tagging/) for adding turn lanes in `Java OpenStreetMap Editor (JOSM)` in a hassle free way during this time. **The plugin intuitively provides presets for the most common, and recently used, turn lane combinations, and also automatically updates lane count based on the number of turn lanes for the road. You can install it from `JOSM Preferences > Plugins > turnlanes-tagging` and restarting your JOSM.** Please make sure that you check out the plugin and provide inputs and feedback on further improvements that can be made to it. 


#### Going forward
The team will continue adding turn lanes in [Denver](http://tasks.openstreetmap.us/project/48#) and [Seattle](http://tasks.openstreetmap.us/project/49#). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by drkludge** [comment]
On Tue, Jun 28, 2016 at 12:37 AM, Abhishek Saikia <notifications@github.com>
wrote:

> We have also introduced an plugin called TurnLanes-tagging plugin
> <https://www.mapbox.com/blog/turnlanes-tagging/> for adding turn lanes in Java
> OpenStreetMap Editor (JOSM) in a hassle free way during this time. *The
> plugin intuitively provides presets for the most common, and recently used,
> turn lane combinations, and also automatically updates lane count based on
> the number of turn lanes for the road. You can install it from JOSM
> Preferences > Plugins > turnlanes-tagging and restarting your JOSM.*
> Please make sure that you check out the plugin and provide inputs and
> feedback on further improvements that can be made to it.

 Abhishek and et all MapBox Staff,

Thank you for these reports.  I understand that they are important to your
work because there is still a large question in some mappers view about
these paid mapper efforts.  It violates the "we always did it this way"
method of doing business.

   - Again thank you very much for your efforts.
   - The efforts are something that I can build on when the time comes.
   - Would you please consider putting your documentation in the tag?
   Right now you are working on turn lanes.  You all have worked through many
   cases that have not been developed well at this point in the current OSM
   wiki page.  However, I do not feel that I can take your github work and put
   the documentation in the wiki.  Where my eyes have glazed over looking at
   the OSM turn lane page before, your work would be a great addition because
   of the clarity brought by a number of edits.  That's something that can be
   built on. You may have to add the work in a section below the current tag
   documentation.
   - Would you please consider other tags that you have worked on in the
   past or future?  Ditto for adding the documentation below the existing page
   doc.
   - Thank you for the JOSM plugin.

I'll have to give these features a whirl when I get a chance.

Regards,
Greg

**Comment by abhisheksaikia** [comment]
Hey @drkludge Thanks for the encouraging feedback and your suggestions. We have already gone ahead and added the `turn lanes tagging` plugin to the [wiki](http://wiki.openstreetmap.org/wiki/Key:turn#Turning_indications_per_lane) page. Currently we are reviewing and working on adding our documentation on turn lanes mapping and other tags we have used during this sprint to the wiki page. This will be done ASAP. Meanwhile please do keep reviewing our tools and keep the feedback coming on our tooling and processes :).

**Comment by drkludge** [comment]
So I am now beginning to believe that there is a certain fragility with OSM relation.  I think turn restrictions, lanes counts, bus routes or routes in general and perhaps a **bug in the knife tool** are at odds with each other.  The problem could be how JOSM handles splitting ways so that you don't end up with a conflict with yourself during a relation change.  The long story short is that the Phoenix/Maricopa County bus routes are being shredded right now with the turn lane changes that are being made right now.  I have a list of the relations so that they can be repaired http://www.openstreetmap.org/way/436781523/history#map=15/33.4972/-112.1005&layers=T .   I will be happy to pitch in and fix these after a way in complete.  I don't know if you have a way of determining the order of ways, etc.

By the way, I think that this is great progress and thank you for your help in the Metro Phoenix area.

The problem almost feels like a whole way with history is deleted and a new way is created in its place.  I wonder if this is the knife tool and JOSM interaction.  Perhaps the knife tool is not considering the relations on a way when it splits a way.  It also feels like JOSM requires a mapper to add the new way segment to replace the one way that existed before it was split in two.  Would you please look into the problem?

Thanks,
Greg

**Comment by dannykath** [comment]
In this [case] (https://www.openstreetmap.org/way/64014597/history)

![example](https://cloud.githubusercontent.com/assets/8483644/17751036/e82360b2-648a-11e6-8919-efe01a9e9aba.png)
**this dots represents a roundabout, usually painted in the way before the roundabout.**

**Comment by RichRico** [comment]
### Chicago task completion summary:
We wrapped up mapping turn lanes in [Chicago](http://tasks.openstreetmap.us/project/50) and added a total of 12,388 turn lanes. The total turn lanes count in Chicago after the sprint is 12,577 at present. Similar to our approach in [San Diego](http://tasks.openstreetmap.us/project/51) we only focused on adding turn lanes to roads classified as secondary and above.

#### Final Statistics in Detroit turn lanes mapping sprint (including validation):

* Total number of working days: **5**
* Existing turn lanes before sprint: **161**
* Total number of turn lanes mapped during sprint: **12,388**
* Total area covered: **1915.60 square kilometers**
* Turn lanes density: **~7 turn lanes/ square kilometer**

#### Observations
* There are on an average of ~7 turn lanes per square kilometer in Chicago which is higher than the turn lanes in Seattle and in other cities that we mapped so far.

* The Satellite imagery coverage in New York is good enough to map turn lanes. We stuck to Bing imagery for the entire task.

#### Going forward
The team will continue adding [turn lanes in San Diego](http://tasks.openstreetmap.us/project/51). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by RichRico** [comment]
### San Diego task completion summary:
We wrapped up mapping turn lanes in [San Diego](http://tasks.openstreetmap.us/project/51) and added a total of 8,869 turn lanes. The total turn lanes count in Chicago after the sprint is 9,627 at present. Similar to our approach in [Austin](http://tasks.openstreetmap.us/project/55) we only focused on adding turn lanes to roads classified as secondary and above.

#### Final Statistics in Detroit turn lanes mapping sprint (including validation):

* Total number of working days: **3**
* Existing turn lanes before sprint: **631**
* Total number of turn lanes mapped during sprint: **8,869**
* Total area covered: **1458.49 square kilometers**
* Turn lanes density: **~7 turn lanes/ square kilometer**

#### Observations
* There are on an average of ~7 turn lanes per square kilometer in SanDiego which is same than the turn lanes in Chicago.

* The Satellite imagery coverage in New York is good, also we guided whith `USGS Large Scale Imagery` to map turn lanes.

#### Going forward
The team will continue adding [turn lanes in Austin](http://tasks.openstreetmap.us/project/55). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by RichRico** [comment]
### Austin task completion summary:
We wrapped up mapping turn lanes in [Austin](http://tasks.openstreetmap.us/project/55) and added a total of 2,178 turn lanes. The total turn lanes count in Austin after the sprint is 2,345 at present. Similar to our approach in [San Diego](http://tasks.openstreetmap.us/project/51) we only focused on adding turn lanes to roads classified as secondary and above.

#### Final Statistics in Austin turn lanes mapping sprint (including validation):

* Total number of working days: **2**
* Existing turn lanes before sprint: **167**
* Total number of turn lanes mapped during sprint: **2,178**
* Total area covered: **1065.09 square kilometers**
* Turn lanes density: **~2 turn lanes/ square kilometer**

#### Observations
* There are on an average of ~2 turn lanes per square kilometer in Austin which is lower than the turn lanes in other cities that we mapped so far.

* The Satellite imagery coverage in Austin is not good, also we guided whith `USGS Large Scale Imagery` to map turn lanes.

#### Going forward
The team will continue adding [turn lanes in Phoenix](http://tasks.openstreetmap.us/project/56). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by karitotp** [comment]
### Phoenix task completion summary:
We wrapped up mapping turn lanes in [Phoenix](http://tasks.openstreetmap.us/project/56) and added a total of 13,841 turn lanes. The total turn lanes count in Phoenix after the sprint is 14,069 at present. Similar to our approach in [Austin](http://tasks.openstreetmap.us/project/55) we only focused on adding turn lanes to roads classified as secondary and above.

#### Final Statistics in Phoenix turn lanes mapping sprint (including validation):

* Total number of working days: **4**
* Existing turn lanes before sprint: **94**
* Total number of turn lanes mapped during sprint: **13,841**
* Total area covered: **2226.58 square kilometers**
* Turn lanes density: **~6 turn lanes/ square kilometer**

#### Observations
* There are on an average of ~6 turn lanes per square kilometer in Phoenix.
* The Satellite imagery coverage in Phoenix is good enough to map turn lanes.

#### Going forward
The team will continue adding [turn lanes in Indianapolis](http://tasks.openstreetmap.us/project/57). Join in if you are interested in mapping or let us know how we can improve the workflow at any time.

**Comment by drkludge** [comment]
Please see my two issues that I have raised.  You need to fix your process before moving to another city!

**Comment by RichRico** [comment]
Hello @drkludge 
Sorry for the delay in the response, our team taking this as a serious issue and looking into it. We will get back to you soon with our finding.

**Comment by samely** [comment]
@drkludge thank you for reviewing and fixing our edits. We identified the broken relation error is not related to the knife tool, as this maintains all relations in the new segment. See bellow.

![anim](https://cloud.githubusercontent.com/assets/10425629/17780694/ecf57600-6531-11e6-95eb-6e5c0fdbf46e.gif)

We also did further analysis on this issue and found that due to the existing of many relations in Phoenix we have conflicts while editing roads that are part of the same relation. As we know [the task manager](http://tasks.openstreetmap.us/project/56) allows us to work in a certain zone at the same time to avoid conflicts in highways, but we didn't consider that [one relation](https://www.openstreetmap.org/relation/4499169) can even cross over more than 10 blocks.

![image](https://cloud.githubusercontent.com/assets/10425629/17780125/e4bc7738-652f-11e6-9887-f4c1e36e5e1e.png)
*Two users made changes with few minutes apart*

You are totally right, we need to improve our process before continue working on `turn:lanes`. If you have a solution for this, please let us know so the process can be improved.

[Ref1](https://github.com/mapbox/mapping/issues/223) [Ref2](https://github.com/mapbox/auto-tools/issues/11)

### Next action
- [ ] @mapbox/team-data stop working on `turn:lanes`
- [ ] @samely and @dannykath obtain a list of broken relations in all cities where turn:lanes were added and fix them.
- [ ] @rub21, @planemad and @andygol  find a way to avoid editing the same relations at the same time.

**Comment by Andygol** [comment]
per chat with @dannykath 

This spreadsheet contains list of `route`=`bus` relations that need to check

**Comment by Andygol** [comment]
per chat with @dannykath 

This table contains list of `route`=`bus` relations that could be affected during working on adding `turn:lanes`. 

https://docs.google.com/spreadsheets/d/1zlkn7IB2QtQwmeE-UBcEoH8_7DD1N2mbaCqyiYC0PdE/edit#gid=343191480

/cc @mapbox/team-data

**Comment by samely** [comment]
Great work @Andygol!

**Comment by Andygol** [comment]
@samely @dannykath  

to get list of relation you can use overpass query similar to this:

```xml
[out:xml][timeout:25];
// fetch area “Philadelphia” to search in
{{geocodeArea:"Philadelphia"}}->.searchArea;
// gather results
(
  relation["type"="route"]["route"="road"](area.searchArea);
);
// print results
out ids;
```

**Comment by drkludge** [comment]
Thank you for looking into to the issue.  I wonder if the Phoenix issue is what happened in Portland Oregon?  Portland is the poster child for a city that uses OSM and has extensive multi-modal routing support in their bus system. You may want to consider other relations.  For example, [USBR 90](http://www.openstreetmap.org/#map=10/33.3099/-111.9946&layers=C) is only 572 miles long.  The one small segment in the southeast corner of the state that bypasses the mountains is as long as Indiana is wide.  I believe Indiana will fit in other main segment 11 times.  The route runs through several cities.  I just looked at [USBR 90 along Union Hills](http://www.openstreetmap.org/#map=16/33.6813/-112.2537&layers=C). It has problems too.  There's a bus route along Union Hills that also must have issues in the same spots.  The challenge with the cycle map is that it renders bi-weekly and sometimes only feels like weekly.  Hence, I am just now seeing the broken route issues in the cycle map.  The bus lines and other highway oriented relationships can have the same lengthy relations too.  I peeked at [AZ 87 that runs along Country Club](http://www.openstreetmap.org/#map=12/33.4057/-111.9139) but did not do a careful investigation of the whole highway route.  The bus route looks okay but I'd have to look at AZ 87 closer.

Based on this review I believe that you will want to have tofix or your tasker look at relationships.  I stopped using tofix along time ago.  tofix has a race condition.  It needs to take an item out of play for 24 hours--the length of time that OSM will keep a change set open.  That's even if someone cannot fix the task.  I think other MapBox editing staff complained about the issue but the author of the code did not understand the problem.

When @samely stated, "...but we didn't consider that one relation can even cross over more than 10 blocks...", now you are beginning to understand the big city/small city divide in OSM.  MapBox and other providers need to enhance where the major markets are at.  However, that leaves out a whole class of potential OSM contributors.  Most of the map designs favor large cities verses small towns.    [Hours of Montana at 80 MPH looks like this picture](http://mapillary.com/map/im/HR11C0Yo2k0pFsQqhhXlFw) verses [hours of California at a snails pace.](http://mapillary.com/map/im/I7Q-DlPswNl_RW1e2mgc_A).  It would be great if I could get a stop sign, yield sign(give way), mile post, or cattle guard(cattle grid) out in the land where towns only have one stop light for the whole city.  Likewise, Phoenix fits into this small town oddity because of the massive urban sprawl that is consumed by Metro Phoenix.  Note that all the tech conventions bypass Phoenix for San Francisco or San Antonio. The active mappers in Phoenix don't run into the editing conflicts that we have seen in this bug because we are so few and far between that I have only had three conflicts in the six plus years of OSM edits.  Your turn lane edits were like we just had a stable base of mappers show up at the same time!  I hope these descriptions of the vastness of the Rocky Mountains helps you understand the problem that you ran into.  I am not surprised.

Would you please make your [Google Doc](https://docs.google.com/spreadsheets/d/1zlkn7IB2QtQwmeE-UBcEoH8_7DD1N2mbaCqyiYC0PdE/edit#gid=343191480) readable to others outside of MapBox?  I am happy to work on the the cleanup because the issue gives me a chance to go back through some of the major streets.  I think that I'll be able to make better headway on the corrections because I understand the routes.

He!He! Did I tell you that I laugh snort when the arm chair mapping page says that I should visit everything I map?  See even these OSM conventions and rhetoric are all geared toward large urban areas that have a stable population.

**Comment by planemad** [comment]
@drkludge appreciate your detailed feedback. We've been looking into this and the problem seems to stem from confusion using JOSM relationship conflict dialog. Opening a new ticket to investigate and track the cleanup.

**Comment by drkludge** [comment]
@mapbox Do you have a better way of getting relation history than I do right now?  I am getting timeouts from OSM.   Here's the reason that I ask.  North and South bound route 7 really got hammered.  Here's the parent relation of the two routes: [4499305](http://www.openstreetmap.org/relation/4499305).  I am thinking that it will be easier for me to get the history of all the route relations.  Then I can edit the .osm file to add the original members back.  Finally, I can add the new segments from the turn lane edits.  In the case of but route 7, one of the child relations was completely cleared out.  I am guessing the empty route is generating the OSM time out.  In the other child relation, only two members are left.  I have the complete list of [Maricopa County Bus route relations in the wiki](http://wiki.openstreetmap.org/wiki/Arizona/Maricopa_County/Public_Transport).  Would you be able to provide me history of these relations before the turn lane mapping started?

Also I repaired the [USBR 90](http://www.openstreetmap.org/changeset/41620747) bike route this morning.

**Comment by drkludge** [comment]
geofabrik.de had an Arizona file back to 8/1/2016.  It looks like MapBox wasn't in the valley until 8/12ish based on the WhoDidIT map and [Project 56](http://tasks.openstreetmap.us/project/56).  That should speed the cleanup.

**Comment by drkludge** [comment]
The good news is that I am able to repair the route with my regex of the osmosis member file.  The even better news is that the bus route looks like it should with missing segments.  The interesting news is that bus route 7 issues occurred before the lane expansion. [Change set 41628004](http://www.openstreetmap.org/changeset/41628004) repaired the north bound route.

**Comment by rickmastfan67** [comment]
Sounds like what happened here is that some of the Mapbox team didn't refresh the data on their end before continuing onto another road to add in turn lanes, thus creating the problems @drkludge mentioned above with the damaged relations.  Looking at [this screenshot](https://cloud.githubusercontent.com/assets/10425629/17780694/ecf57600-6531-11e6-95eb-6e5c0fdbf46e.gif) that @samely posted above, it seems you aren't taking advantage of a major item in JOSM that helps to try to prevent stuff like this.  The "Update Modified" tool (which I've personally pinned right beside the 'upload' button in the toolbar, Ctrl+Alt+M should be the shortcut for it) is something I ALWAYS use before even attempting to upload anything to the OSM database to help prevent any possible chance of a conflict.  Normally, I never do get a conflict, but in the rare times I have, this has saved me from screwing anything up.  And if what I did was a minor change, I just delete the layer, and restart with fresh data to prevent any problems.  Otherwise, I do attempt to solve the conflict right there before attempting to upload.

Another tactic I've been using is whenever I have finished uploaded something (after of course doing the "Update Modified" before the upload), I always completely delete the layer I was working on, and download a completely fresh copy of the area data before I continue on editing.  This should always be done between uploads, and really should become second nature to ANY experienced OSM mapper.  This also helps prevents any problems when somebody else has uploaded a new version of a way/relation that they split since you downloaded the data from the OSM servers.  For example, if somebody had just created a new relation that covers two ways, but it isn't in your copy of the data, and you split one of the ways in that relation?  That's a great example of how the data can easily get broken and possibly create unintended gaps in relations that shouldn't be there.  Sure, it might slow you down by 20 seconds or so between tasks, but would you rather spend an hour or more fixing relation data that got damaged because of not keeping the data synced on your end?  I'd chose the 20 seconds over the hour myself. lol.

**Comment by planemad** [comment]
@drkludge @Andygol @samely lets move the broken relation conversation to https://github.com/mapbox/mapping/issues/224

**Comment by jothirnadh** [comment]
Recently Mapbox satellite imagery in DC is rendered with [2016 ortho imagery](https://www.mapbox.com/blog/washington-aerial-imagery/) which will be a great spot to resume our turn lane mapping effort which we paused in August 2016. Created a [turn lane mapping task for DC](http://tasks.openstreetmap.us/project/74) which we can target to hit next week.

### Points to be noted:  
* Let's make sure to add missing streets at the junctions where we map turn lanes if we come across any 🙏 .
* Make sure the relation of the road is not broken before splitting the road for adding a turn lane
* Let's stick to the [same workflow](https://github.com/mapbox/mapping/issues/180#issue-152963087) which we followed to map turn lanes in Phoenix apart from adding missing streets at junctions.

cc:// @RichRico @maning @abhisheksaikia

**Comment by Andygol** [comment]
There is data set with **Intersection Points** on opendata.dc.gov with 18,276 features - http://opendata.dc.gov/datasets/4b4155ee7bb443aa9a37a3f8e22a8abc_35
<img width="1203" alt="intersection points 2017-01-20 19-12-10" src="https://cloud.githubusercontent.com/assets/369696/22158400/b82e3bea-df44-11e6-9f3b-d6597fae691a.png">

There is a possibility to mark `turn:lanes` on the roads around ☝️ these points. For this purpose it is good to use https://osmlab.github.io/to-fix/.

@Rub21, is it possible to make `to-fix` task for [one](https://www.openstreetmap.org/relation/4631298) of DC neighborhoods for trial run?

**Comment by Andygol** [comment]
Doing  **turn lane mapping task for DC** please pay attention on such cases 👇 
and apply right `oneway=yes` tagging and direction for the way

<img width="752" alt="java openstreetmap editor 2017-01-20 20-24-51" src="https://cloud.githubusercontent.com/assets/369696/22160673/4d4562c6-df4f-11e6-9a66-ea85332e6b93.png">

**Comment by maning** [comment]
>  is it possible to make to-fix task for one of DC neighborhoods for trial run?

@Andygol before we do this, some clarifications;

* The way to-fix works is that a mapper will be assigned a random intersection to work for every task.  We lose context on the area.
* We may introduce conflicts when several people were assigned to an intersection that is part of a long road.
* We don't map turn lanes on residential roads, the intersection you linked include all intersections regardless of highway type.  How do we filter this out?

@jothirnadh @Rub21 @samely

**Comment by Andygol** [comment]
@maning 

> * The way to-fix works is that a mapper will be assigned a random intersection to work for every task.  We lose context on the area.

Only in case if we add not only `turn:lanes` but we add `lanes=*` tagging along the way between intersection

> * We may introduce conflicts when several people were assigned to an intersection that is part of a long road.

The same may happened even in standard workflow when way goes across two tasks 

> * We don't map turn lanes on residential roads, the intersection you linked include all intersections regardless of highway type.  How do we filter this out?

There is possible just to skip it as we do it in `to-fix` usually or before creating task in `to-fix` find nodes that is near **Major** roads 

cc @jothirnadh @Rub21 @samely

**Comment by Rub21** [comment]
@Andygol @maning :hand: 
> There is possible just to skip it as we do it in to-fix usually or before creating task in to-fix find nodes that is near Major roads

I can help you filtering this data for Major Roads, 18,276  is a lot of items  to do a  skip.

**Comment by jothirnadh** [comment]
@RichRico @Rub21 @samely it would be great if @mapbox/peru-office can take up this [DC turn lane mapping](http://tasks.openstreetmap.us/project/74) task and target to complete it as soon as possible 🚀

**Comment by Rub21** [comment]
@Andygol @maning , I just filtered the point for major roads, it reduces to 10,198 items.  find all here, https://gist.github.com/anonymous/268daa5bd7f747c21d45a4cd1253a4f6.

**Comment by micahcochran** [comment]
The **knife tool** is now part of the **auto_tools** plugin, which can easily be installed through the default plugin installation.  No "activation" needed.
See  http://wiki.openstreetmap.org/wiki/JOSM/Plugins/Knife-tool

**Comment by RichRico** [comment]
@jothirnadh @Andygol 
Today the ayacucho office began to work on the [task](http://tasks.openstreetmap.us/project/74) of turn lanes in Washington DC

**Comment by jothirnadh** [comment]
While validating turn lanes in DC, I found that the base data of DC is not aligning with new Mapbox 2015 imagery. Mostly in case of Major highways which are constructed recently.

![screen shot 2017-01-30 at 2 59 22 pm](https://cloud.githubusercontent.com/assets/8401827/22418296/714cf6dc-e6fe-11e6-8d09-1adaff1400e7.png)

This might be because the existing data is traced from outdated Bing imagery (2011) which drastically changed in time.

![mapbox-bing change](https://cloud.githubusercontent.com/assets/8401827/22417890/7a1aef1e-e6fc-11e6-9e82-e1b253edba4f.gif)

*Mapbox-2015, bing-2011*
 
It would be great if we can have a separate task to align major roads in DC city with Mapbox 2015 imagery to update the base map.

cc:// @maning @planemad @abhisheksaikia @RichRico @shvrm

**Comment by oini** [comment]
> It would be great if we can have a separate task to align major roads in DC city with Mapbox 2015 imagery to update the base map.

@jothirnadh Correct me if I'm wrong @maning @bkowshik, but I think missing streets in DC will be detecting these cases 🤔

**Comment by RichRico** [comment]
#### Phoenix task completion summary:

Great work @mapbox/team-data on adding and reviewing turn restrictions in Washington DC 🎉 . Our team reviewed and added turn lanes based on the new [3-inch recent orthoimagery](https://www.mapbox.com/blog/washington-aerial-imagery/) in DC. Do check it out.

During the course of this task, our team added **~2548** turn lanes in DC during a course of **3 mapping days** with a total of **8 mappers** working on the task. The total count of turn lanes in DC now is ~**3200**.

#### Final Statistics in Washington DC turn lanes mapping sprint (including validation):
- Total number of working days: **3**
- Existing turn lanes before sprint: **806**
- Total number of turn lanes mapped durg sprint: **2548**
- Total area covered: **755.399 square kilometers**
- Turn lanes density: **~3 turn lanes/ square kilometer**

### Findings from the current task:
- The major highways are not [aligned with Mapbox imagery in core city area](https://github.com/mapbox/mapping/issues/180#issuecomment-276020440) which should be fixed to improve base map in DC.
- The existing road classification in DC looks a bit odd, should be cross verified once.
- In general, the road with physical separation is mapped as a dual carriageway in OSM. While mapping turn lanes if we come across any roads mapped as single carriageway instead of dual carriageways we make sure to split the road. But if that road is having an existing relation, then we skip the road to make sure we don't break the relation. We are coming across such cases more often and it is good to get a solution for this problem. @andygol and @samely and ideas on how we can solve this issue.
- Turn lanes in tertiary roads are not being mapped as a part of the current workflow but found a couple of them existing in tertiary roads. Continuing from next task lets make sure to cross verify the ratio of turn lanes on tertiary roads before starting the task.
- In this particular task turn lanes were mapped using Mapbox imagery for core city area and Bing imagery at city outskirts, going forward the task can be limited so the people don't need to cross verify with varying sources.

Please feel free to contribute and give feedback on our mapping processes on our [mapping repo](https://github.com/mapbox/mapping/) where we track all our ongoing mapping projects.

**Comment by jothirnadh** [comment]
Great work in mapping turn lanes in DC 👏 . Continuing turn lane mapping quest, let's hit missing turn lanes in outskirts of LA covering Pomona, San Bernardino, and Redlands. [Created a mapping task for this region](http://tasks.openstreetmap.us/project/81) which has **~582** existing turn lanes.

### Major blockers:
* There are a lot of roads mapped as bi-directional roads, which are supposed to be mapped as dual carriageways, due to the presence of physical barrier in between them. Please make sure to split them into dual carriageways before mapping.
* Don't split the roads having route relations as there is no proper workflow to fix them. Need help from local community members for fixing this issue.

Ex: 
![screen shot 2017-02-13 at 5 20 04 pm](https://cloud.githubusercontent.com/assets/8401827/22882362/bd8fa5b8-f210-11e6-9a74-394cf195329a.png)

[Road mapped as bi-directional road with an existing relation](https://www.openstreetmap.org/way/342638965/history)

cc:// @mapbox/team-data

**Comment by Andygol** [comment]
You can find the description of the workflow "How to split a bidirectional road to the road with separate ways in each direction" here - <https://gist.github.com/Andygol/a2ff609bb4d71274245d51b1c253b832>

![](https://cloud.githubusercontent.com/assets/369696/22930589/a3ee8f3e-f2c9-11e6-8ad9-19fd824a19c7.gif)
_dealing with relations ☝️_

**Comment by Andygol** [comment]
@mapbox/team-data if you are faced relation for a Public Transport route, ping me with links to it for extending workflow on such case.

**Comment by jothirnadh** [comment]
Thanks for sharing the ^^ workflow @Andygol, @chtnha and I will do a trial run and post our findings soon before including it into our regular workflow 🙇 .

### Major findings from validation so far:
* We are not mapping turn lanes/ splitting the road as dual carriageways if the road has an existing`Route relations`, but finding roads not touched even if it is tagged with turn restrictions. Let's make sure the roads with turn restrictions are not skipped.
* Don't touch roads with `route relation` for now, don't even map turn lanes in that roads.
* In few blocks the bi-directional roads with physical separation are not mapped as dual carriageways even if they are not a part of any route relation.

@mapbox/team-data do invalidate the blocks which fail to fulfill the above criteria and let the mapper fix them.

**Comment by ridixcr** [comment]
Guys when I was validating turn lanes on [# 81 - Turn Lanes in Los Angeles outskirts- Chino, San Bernardino & Redlands](http://tasks.openstreetmap.us/project/81#task/327) found this error and solved it, as follow, according to the `turn restriction` rules.

![Gif](https://cloud.githubusercontent.com/assets/1646072/23041776/23a69d8e-f464-11e6-855e-f38f14378c05.gif)

@Andygol @jothirnadh @RichRico

**Comment by Yurasi** [comment]
> @mapbox/team-data do invalidate the blocks which fail to fulfill the above criteria and let the mapper fix them.

@mapbox/team-data, we shouldn't invalidate the blocks but correct them, in this way we aren't going to waste time in review by one person and correction by another person.

### Some findings from validation so far:

- The number of lanes not coincide with number of carriles. This because of that some aren't using the turnlanes plugin.
- Some  make a mistake of identify merge_to with slight. 

@jothirnadh

**Comment by RichRico** [comment]
### Turn Lane LA exterior [task](http://tasks.openstreetmap.us/project/81) completion summary:

Great work @mapbox/team-data on adding and reviewing turn lanes in LA Exterior 🎉 . that were missing turn lanes in outskirts of LA covering Pomona, San Bernardino, and Redlands.

During the course of this task, our team added **~9,075** turn lanes in LA Exterior during a course of **5 mapping days** with a total of **22 mappers** working on the task. The total count of turn lanes in LA exterior now is ~**9,657**.

#### Final Statistics in Washington DC turn lanes mapping sprint (including validation):
- Total number of working days: **5**
- Existing turn lanes before sprint: **582**
- Total number of turn lanes mapped durg sprint: **9,075**

**Comment by maning** [comment]
Paused.  Wil  revisit later.
